generate_latest_json.py

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env python3
+"""
+Sopel update-check JSON generation utility
+
+This script creates a `latest.json` file representing the most recent stable and
+unstable `sopel` releases published to PyPI, using PyPI's own JSON API.
+
+Copyright 2024 dgw, technobabbl.es
+Licensed under the Eiffel Forum License 2.
+
+https://sopel.chat
+"""
+import json
+
+from packaging.version import parse as parse_version
+import requests
+
+
+def main(argv=None):
+    data = requests.get('https://pypi.org/pypi/sopel/json').json()
```

Comment:
Let's make the error a little more obvious if something goes wrong with the request

```suggestion
    response = requests.get('https://pypi.org/pypi/sopel/json')
    response.raise_for_status()
    data = response.json()
```

Reply:
This is fair, and I'm going to commit the suggestion.

Bigger picture: Error handling is not the best in _any_ of these helper scripts, but only this one actually makes a network request. I haven't thought of a way to, say, cache the last known set of releases in case the request fails. I'd kind of like the site build to not fail just because PyPI hiccups, but that might be too rare of an occurrence to spend dev time guarding against.